src/pybkgmodel/data.py

```diff
@@ -501,23 +537,236 @@ def load_events(cls, file_name, cuts):
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        int
+            Observation ID number of the fits file
+
+        Raises
+        ------
+        RuntimeError
+            In case Observation ID is not found or Events HDU layer is missing. In this case the file is not complient to GADF.
+        """
+
+        obs_id = None
+
+        with fits.open(file_name, memmap=False) as input_file:
+            try:
+                obs_id = int(input_file["EVENTS"].header['OBS_ID'])
+            except Exception as error:
+                raise RuntimeError(f'Can not find observations ID in {file_name}') from error
+
+        return obs_id
+
+    @classmethod
+    def load_events(cls, file_name):
+        """Function to read the events from the DL3 file and to compute the missing variables needed by pybkgmodel
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        dict:
+            A dictionary with the event properties: arrival time, direction, and energy.
+        """
+
+        data_names_mapping = {
+            'EVENT_ID': 'daq_event_number',
+            'RA': 'event_ra',
+            'DEC': 'event_dec',
+            'GAMMANESS': 'gammaness',
+            'ENERGY': 'event_energy',
+        }
+
+        with fits.open(file_name, memmap=False) as input_file, \
+            erfa_astrom.set(ErfaAstromInterpolator(1 * u.s)):
+            try:
+                evt_hdu = input_file["EVENTS"]
+                evt_head = evt_hdu.header
+                evt_data = pandas.DataFrame(evt_hdu.data)
+
+                event_data = {}
+
+                for key, name in data_names_mapping.items():
+
+                    if key in evt_data.keys():
+                        event_data[name] = evt_data[key].to_numpy()
+                        # Fits header keywords start counting from 1
+                        unit_name = f"TUNIT{evt_data.columns.get_loc(key)+1}"
+
+                        try:
+                            event_data[name] *= u.Unit(evt_head[unit_name])
+                        except KeyError:
+                            event_data[name] *= u.one
+
+                # Event times need to be converted from Instrument reference epoch
+                ref_epoch = astropy.time.Time(evt_head['MJDREFI']+evt_head['MJDREFF'], format='mjd')
+
+                event_data['mjd'] = astropy.time.Time((evt_data['TIME'].to_numpy() 
+                                                       + ref_epoch.unix),
+                                                      scale='utc',
+                                                      format='unix'
+                                                      ).mjd * u.d
+
+                # Adapt
+                evt_time = astropy.time.Time(event_data['mjd'], format='mjd')
```

Comment:
now you create a time object a second time.

Just to `evt_time = ref_epoch + evt_data["TIME"].quantity` and then `evt_data["mjd"] = evt_time.mjd`

Reply:
All of this is also already implemented in Gammapy, e.g. here: https://github.com/gammapy/gammapy/blob/main/gammapy/utils/time.py

Reply:
Simplified according to your suggestion.